library/src/main/java/com/mapzen/android/MapzenMap.java

```diff
@@ -676,4 +708,26 @@ public void drawTransitRouteLine(@NotNull List<LngLat> points, @Nullable List<Ln
   public void clearTransitRouteLine() {
     overlayManager.clearTransitRouteLine();
   }
+
+  /**
+   * By default all {@link MapData} objects are removed from the map when it is destroyed. To
+   * persist this data such as in the case of orientation changes, use this method.
+   * @param persistOnRecreation persist {@link MapData} across orientation changes
+   */
+  public void setPersistMapData(boolean persistOnRecreation) {
+    overlayManager.setPersistMapData(persistOnRecreation);
+    mapStateManager.setPersistMapState(persistOnRecreation);
+  }
+
+  private void restoreMapState() {
```

Comment:
I would wager `setStyle(MapStyle)` is what's slow. `MapController$loadSceneFile(String)` is a heavy, _heavy_, call (this is why we have a method to do it on a background thread). We have some ideas about speeding this up for large scene files in the future, but right now it's definitely worthwhile to do whatever you can to avoid this call. Maybe we could check whether the 'persisted' style is the same as the current style?